atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/DefaultClusterEventService.java

```diff
@@ -702,58 +418,34 @@ TopicIterator iterator() {
      *
      * @param subscription the subscription to add
      */
-    synchronized void addRemoteSubscription(final InternalSubscriptionInfo subscription) {
+    synchronized void addRemoteSubscription(final MemberId subscription) {
```

Comment:
Do you know why this is synchronized? Is this really necessary?

Reply:
I would assume not, since we are using here a concurrent hash set.

Reply:
You are right. Before it was not a concurrent set.